cirq/google/programs.py

```diff
@@ -28,25 +28,78 @@
 
 def gate_to_proto_dict(gate: ops.Gate,
                        qubits: Tuple[ops.QubitId, ...]) -> Dict:
-    xmon_gate = xmon_gate_ext.try_cast(  # type: ignore
-        xmon_gates.XmonGate, gate)
-    if xmon_gate is not None:
-        return xmon_gate.to_proto_dict(*qubits)
-
     if isinstance(gate, ops.MeasurementGate):
         return _measure_to_proto_dict(gate, qubits)
+
+    if isinstance(gate, ops.XPowGate):
+        if len(qubits) != 1:
+            # coverage: ignore
+            raise ValueError('Wrong number of qubits.')
+        return _x_to_proto_dict(gate, qubits[0])
+
+    if isinstance(gate, ops.YPowGate):
+        if len(qubits) != 1:
+            # coverage: ignore
+            raise ValueError('Wrong number of qubits.')
+        return _y_to_proto_dict(gate, qubits[0])
+
+    if isinstance(gate, (xmon_gates.ExpWGate, ops.PhasedXPowGate)):
+        if len(qubits) != 1:
+            # coverage: ignore
+            raise ValueError('Wrong number of qubits.')
+        return _phased_x_to_proto_dict(gate, qubits[0])
+
     if isinstance(gate, ops.ZPowGate):
         if len(qubits) != 1:
+            # coverage: ignore
             raise ValueError('Wrong number of qubits.')
         return _z_to_proto_dict(gate, qubits[0])
+
     if isinstance(gate, ops.CZPowGate):
         if len(qubits) != 2:
+            # coverage: ignore
             raise ValueError('Wrong number of qubits.')
         return _cz_to_proto_dict(gate, *qubits)
 
     raise ValueError("Don't know how to serialize this gate: {!r}".format(gate))
 
 
+def _x_to_proto_dict(gate: ops.XPowGate, q: ops.QubitId) -> Dict:
+    exp_w = {
+        'target': cast(devices.GridQubit, q).to_proto_dict(),
+        'axis_half_turns':
+            xmon_gates.XmonGate.parameterized_value_to_proto_dict(0),
```

Comment:
Move parameterized_value_to_proto method to the module level

Reply:
I don't think we should do that quite yet. In particular, I think we want one single "give me a proto" method instead of a family of them. Perhaps a magic method protocol...